src/finn/custom_op/fpgadataflow/elementwise_binary.py

```diff
@@ -0,0 +1,809 @@
+        # Set the new output data type as attribute
+        self.set_nodeattr("out_dtype", out_dtype.name)
+        # Annotate the output tensor with the new data type
+        model.set_tensor_datatype(self.onnx_node.output[0], out_dtype)
+        # Return the minimized output data type
+        # Note: Probably not required by MinimizeAccumulatorWidth transformation
+        return out_dtype
+
+    # Derives the optimal width of the output data type
+    def _derive_out_dtype(self, model: ModelWrapper):
+        # Depends on the actual operation performed and must be specialized by
+        # the concrete implementations
+        raise NotImplementedError(
+            f"_derive_out_dtype of {self.__class__.__name__}"
+            f" is not implemented!"
+        )
+
+    # Minimizes the width of the weight data type, 'weight' here due to
+    # convention, it actually applies to any constant initializer input
+    def minimize_weight_bit_width(self, model: ModelWrapper):
+        # Check for an initializer providing the left hand side input
+        lhs = model.get_initializer(self.onnx_node.input[0])
+        # If the left hand side input is provided as initializer, minimize the
+        # bits used for storing this
+        if lhs is not None:
+            # Remember the "style" of receiving the input for further code
+            # generation
+            self.set_nodeattr("lhs_style", "const")
+            # Minimum and maximum "weight" on the left hand side, determining
+            # the range of values which needs to be represented
+            _min = lhs.min()
+            _max = lhs.max()
+            # Determine whether signed or unsigned type is required for
+            # representing the weights and select the largest "signed magnitude"
+            _mag = _max if _min > 0 else \
+                _min if (abs(_min) > _max) else (-_max - 1)
+            # Smallest data type large enough to represent this range of values
+            dtype = DataType.get_smallest_possible(_mag)
+            # Update the corresponding data type attribute of the node
+            self.set_nodeattr("lhs_dtype", dtype.name)
+            # Annotate the tensor with the new data type
+            model.set_tensor_datatype(self.onnx_node.input[0], dtype)
+
+        # Check for an initializer providing the right hand side input
+        rhs = model.get_initializer(self.onnx_node.input[1])
+        # If the right hand side input is provided as initializer, minimize the
+        # bits used for storing this
+        if rhs is not None:
+            # Remember the "style" of receiving the input for further code
+            # generation
+            self.set_nodeattr("rhs_style", "const")
+            # Minimum and maximum "weight" on the right hand side, determining
+            # the range of values which needs to be represented
+            _min = rhs.min()
+            _max = rhs.max()
+            assert _min != 0
+            assert _max != 0
```

Comment:
why is there an assert for min/max of rhs not equal to zero, specifically for const rhs? 
this can happen for e.g. ReLU implemented as ElementwiseMaximum(x, 0)